### PR TITLE
Fix retain cycle in LiveViewCoordinator

### PR DESCRIPTION
**Pull request by nighthawk**
`LiveViewCoordinator` is initialised from `LiveSessionCoordinator`, which gets passed in as a parameter to `init` and then retained by the view coordinator. This causes a retain cycle.

Additionally, this is also retained by the `handleEvent` as `session` was used directly in the closure, which then retained it.

This commit fixes both by using a weak reference from the view coordinator to the session coordinator, and accessing the session via the weak `self` reference.

### CODE REVIEW

**Comment by nighthawk** [comment]
The retain cycle is triggered by `LiveSessionCoordinator`'s `init` method which has this line:

```swift
self.navigationPath = [.init(url: url, coordinator: .init(session: self, url: self.url))]
```

Note how `self` is passed there, which causes the retain cycle.

I noticed this when I didn't have my local server running and the LiveView in the app couldn't connect, and when I navigated away from the LiveView in my app and it was no longer in the view hierarchy, the live view session kept trying to reconnect.

**Comment by nighthawk** [comment]
Seems like that retain cycle hid two other issues here. Since `LiveSessionCoordinator` will now get deallocated properly, the two `unused self` usages need to be removed as they can fire in rare cases after it got deallocated, which then crashes.

**Review by carson-katri — CHANGES_REQUESTED** [review]
Great catch, thanks!

Could you update the CHANGELOG.md file to note this fix?

**Comment by bcardarella** [comment]
This should go under `Fixed` in the `CHANGELOG.md`

File: Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift [hunk 1]
```diff
@@ -32,7 +32,7 @@ public class LiveViewCoordinator<R: RootRegistry>: ObservableObject {
         internalState
     }
     
-    @_spi(LiveForm) public let session: LiveSessionCoordinator<R>
+    @_spi(LiveForm) public weak var session: LiveSessionCoordinator<R>!
```
Comment (carson-katri) [inline Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift:35]: This should be `private(set)`
```suggestion
    @_spi(LiveForm) public private(set) weak var session: LiveSessionCoordinator<R>!
```

**Comment by nighthawk** [comment]
Sure thing. Change log updated.

**Review by carson-katri — APPROVED** [review]
LGTM 👍

**Comment by bcardarella** [comment]
@nighthawk thank you!

**Pull request merged**